expr/func_aspercent_test.go

```diff
@@ -555,6 +563,28 @@ func testAsPercent(name string, in []models.Series, out []models.Series, t *test
 			t.Fatalf("case %q (%q, %v): output point %d - expected %v got %v", name, total, nodes, j, p, g.Datapoints[j])
 		}
 	}
+
+	// Test if original series was modified
+	for i, orig := range originalSeries {
+		inSerie := in[i]
+		if orig.Target != inSerie.Target {
```

Comment:
Could you use something like https://github.com/grafana/metrictank/blob/0428430b501132244d3a4e94ba8e280c640b07b9/expr/plan_test.go#L186 here?

Reply:
No, because the original series does get modified by metrictank (a "name" tag gets added). Makes sense to compare relevant values only.

Reply:
Scratch that. the real problem was that `reflect.DeepEqual(math.NaN(), math.NaN()) == false`, which is not the case when comparing series